src/shims/unix/sync.rs

```diff
@@ -364,25 +364,32 @@ fn cond_clock_offset<'tcx>(ecx: &MiriInterpCx<'tcx>) -> u64 {
     offset
 }
 
+#[derive(Debug)]
+/// Additional data that we attach with each cond instance.
+struct AdditionalCondData {
+    /// The address of the cond.
+    address: u64,
```

Comment:
Please also store the clock kind in here instead of in-memory, similar to what you did with the mutex kind.